File: frontend/app/src/shared/utils/location.ts
import { LocationDTO } from '@rebrowse/types';

export const readableLocation = (location: LocationDTO) => {
  const locationSegments = [];
  if (location.city) {
    locationSegments.push(location.city);
  }

  if (location.regionName !== location.city) {
    locationSegments.push(location.regionName);
  }

  if (location.countryName) {
    locationSegments.push(location.countryName);
  }

  if (locationSegments.length === 0) {
    return 'Unknown location';
  }

  return locationSegments.join(', ');
};

File: backend/session/session-model/src/main/java/com/rebrowse/location/model/Location.java
package com.rebrowse.location.model;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.NoArgsConstructor;
import lombok.Value;

@Value
@NoArgsConstructor(force = true, access = AccessLevel.PRIVATE)
@AllArgsConstructor
@Builder
public class Location implements Located {

  String ip;
  String countryName;
  String regionName;
  String city;
  String zip;
  String continentName;
  double latitude;
  double longitude;
}
